src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -307,1232 +262,96 @@ object Parser {
-        exactWord("uintptr") ~ not("(" | ".") ^^^ PUIntPtr()
-
-    private def exactWord(s: String): Regex = ("\\b" ++ s ++ "\\b").r
-
-    lazy val qualifiedType: Parser[PDot] =
-      declaredType ~ ("." ~> idnUse) ^^ PDot
-
-    lazy val declaredType: Parser[PNamedOperand] =
-      idnUse ^^ PNamedOperand
-
-    lazy val literalType: Parser[PLiteralType] =
-      sliceType |
-        arrayType |
-        implicitSizeArrayType |
-        mapType |
-        structType |
-        qualifiedType |
-        ghostTypeLit |
-        declaredType
-
-    lazy val implicitSizeArrayType: Parser[PImplicitSizeArrayType] =
-      "[" ~> "..." ~> "]" ~> typ ^^ PImplicitSizeArrayType
-
-
-    /**
-      * Misc
-      */
-
-    lazy val receiver: PackratParser[PReceiver] =
-      "(" ~> maybeAddressableIdnDef.? ~ methodRecvType <~ ")" ^^ {
-        case None ~ t => PUnnamedReceiver(t)
-        case Some((name, addressable)) ~ t => PNamedReceiver(name, t, addressable)
-      }
-
-    lazy val signature: Parser[(Vector[PParameter], PResult)] =
-      parameters ~ result
-
-
-    lazy val result: PackratParser[PResult] =
-      parameters ^^ PResult |
-        typ ^^ (t => PResult(Vector(PUnnamedParameter(t).at(t)))) |
-        success(PResult(Vector.empty))
-
-    lazy val parameters: Parser[Vector[PParameter]] =
-      "(" ~> (parameterList <~ ",".?).? <~ ")" ^^ {
-        case None => Vector.empty
-        case Some(ps) => ps
-      }
-
-    lazy val parameterList: Parser[Vector[PParameter]] =
-      rep1sep(parameterDecl, ",") ^^ Vector.concat
-
-    lazy val parameterDecl: Parser[Vector[PParameter]] = {
-      val namedParam = rep1sep(idnDef, ",") ~ "...".? ~ typ ^^ {
-        case ids ~ variadicOpt ~ t =>
-          ids map { id =>
-            val typ = if (variadicOpt.isDefined) PVariadicType(t.copy) else t.copy
-            PNamedParameter(id, typ).at(id)
-          }
-      }
-      val unnamedParam = ("...".? ~ typ) ^^ {
-        case variadicOpt ~ t =>
-          val typ = if (variadicOpt.isDefined) PVariadicType(t) else t
-          Vector(PUnnamedParameter(typ).at(t))
-      }
-
-      ghostParameter | namedParam | unnamedParam
+    def parse_LL(rule : => Rule): ParserRuleContext = {
+      // thrown by ReportFirstErrorStrategy
+      tokens.seek(0)
+      // rewind input stream
+      reset()
+      val ll_errors = ListBuffer.empty[ParserError]
+      removeErrorListeners()
+      // For full analysis, we want to get all errors.
+      addErrorListener(new InformativeErrorListener(ll_errors, source))
+      setErrorHandler(new DefaultErrorStrategy)
+      // full now with full LL(*)
+      getInterpreter.setPredictionMode(PredictionMode.LL)
+
+      val res = try time ("ANTLR_PARSE_LL", source.name) { rule }
+      catch {
+        case e : Exception => errors.append(ParserError(e.getMessage, Some(SourcePosition(source.toPath, 0, 0))))
+          new ParserRuleContext()
+      }
+      if (ll_errors.isEmpty) errors.clear()
+      res
```

Comment:
is this standart or is anything special happening here? If it is just standard, then please add this as a comment.